vanilla-todo-app/index.js

```diff
@@ -22,10 +22,17 @@ const eTODOActionBtnMode = {
     ADD:'ADD',
 }
 
+function uuidv4() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+      var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+      return v.toString(16);
+    });
+}
+
 class TODOManager{
     constructor(){
         this.todos = [];
-        this.state = { TODOBtnMode: eTODOActionBtnMode.ADD, id:null};
+        this.state = { TODOBtnMode: eTODOActionBtnMode.ADD, id:0};
```

Comment:
now that `id` is a guid, 0 is not a valid value.

Reply:
maybe i just need to delete this useless property?

Reply:
we didn't talk about it :) so the first comment (change to 0) was when it was of type number. now that it's guid, `null` is ok. in general. I think it should be named 'selectedTaskId` and then give it the default `null` value